### ISSUE TITLE
Generic rx.core.typing.Observable[T] does not include .pipe()

### ISSUE THREAD

**Issue by YodaEmbedding** [ReactiveX/RxPY#540]
For example, using `.pipe`:

```python
from rx import Observable
from rx.core.typing import Observable as ObservableT

def func(observable: ObservableT[int]):
    return observable.pipe()
```

`mypy` complains with:

```
test.py:5: error: "Observable[int]" has no attribute "pipe"
Found 1 error in 1 file (checked 1 source file)
```

**Comment by oleic** [comment]
I have the same problem with `Subject`. pyright and mypy are throwing following error:

```
Cannot access member "pipe" for type "Subject[ParametersFrame, ParametersFrame]"
  Member "pipe" is unknown
```

**Comment by dbrattli** [comment]
This will be fixed by #597 that makes `rx.Observable` generic.
